core/src/test/java/io/aiven/klaw/TopicRequestValidatorImplIT.java

```diff
@@ -0,0 +1,276 @@
+    Integer tenantId = 101;
+    int teamId = 1001;
+    Topic topic = utilMethods.getTopic("testtopic");
+    topic.setEnvironment("2");
+    topic.setTeamId(teamId);
+    Env env = utilMethods.getEnvLists().get(0);
+
+    TopicRequestModel addTopicRequest = utilMethods.getTopicRequestModel(1001);
+    addTopicRequest.setEnvironment("2");
+    when(commonUtilsService.isNotAuthorizedUser(any(), any())).thenReturn(false);
+    when(topicControllerService.getUserName()).thenReturn("superadmin");
+    when(topicControllerService.getTeamId(anyString())).thenReturn(teamId);
+    when(topicControllerService.getEnvsFromUserId(any())).thenReturn(List.of("1", "2"));
+    when(commonUtilsService.getTenantId(any())).thenReturn(tenantId);
+    when(topicControllerService.getTopicFromName(anyString(), anyInt()))
+        .thenReturn(Collections.emptyList());
+    when(topicControllerService.getSyncCluster(anyInt())).thenReturn("1");
+    when(mailService.getEnvProperty(anyInt(), anyString())).thenReturn("1,2");
+    when(topicControllerService.getEnvDetails(anyString())).thenReturn(env);
+
+    Set<ConstraintViolation<TopicRequestModel>> violations = validator.validate(addTopicRequest);
+    assertThat(violations.size()).isEqualTo(1);
+    assertThat(violations.toString())
+        .contains("Failure. Please request for a topic first in " + env.getName() + " cluster.");
+
+    when(topicControllerService.getTopicFromName(anyString(), anyInt())).thenReturn(List.of(topic));
+    when(topicControllerService.getEnvDetails(anyString())).thenReturn(null);
+    violations = validator.validate(addTopicRequest);
+    assertThat(violations.size()).isEqualTo(1);
+    assertThat(violations.toString()).contains("Failure. Base cluster is not configured.");
+
+    when(topicControllerService.getEnvDetails(anyString())).thenReturn(env);
+    violations = validator.validate(addTopicRequest);
+    assertThat(violations.size()).isEqualTo(1);
+    assertThat(violations.toString())
+        .contains("Failure. This topic does not exist in " + env.getName() + " cluster.");
+  }
+
+  @Test
+  @Order(7)
+  public void isValidTestValidateTopicConfigParams() {
+    Integer tenantId = 101;
+    Env env = utilMethods.getEnvLists().get(0);
+    String topicPrefixSuffix = "devkafka";
+    env.setOtherParams("topic.prefix=" + topicPrefixSuffix);
+
+    TopicRequestModel addTopicRequest = utilMethods.getTopicRequestModel(1001);
+    when(commonUtilsService.isNotAuthorizedUser(any(), any())).thenReturn(false);
+    when(topicControllerService.getUserName()).thenReturn("superadmin");
+    when(topicControllerService.getEnvsFromUserId(any())).thenReturn(List.of("1"));
+    when(commonUtilsService.getTenantId(any())).thenReturn(tenantId);
+    when(topicControllerService.getEnvDetails(anyString())).thenReturn(env);
+
+    Set<ConstraintViolation<TopicRequestModel>> violations = validator.validate(addTopicRequest);
+    assertThat(violations.size()).isEqualTo(1);
```

Comment:
```suggestion
    assertThat(violations).hasSize(1);
```
assertJ has `hasSize` it's better to use it here

Reply:
Agree, updated.